### DISCUSSION TITLE
Folia

### DISCUSSION THREAD

**Discussion by joaoespanhol** [PurpurMC/Purpur#1311] [__Dismissed__] [answered]
https://github.com/PaperMC/Folia/tree/master/patches/server

nothing like adding multi threads
Improves server performance :D

**Comment by SrBedrock** [comment]
I believe it would be necessary to fork Folia into a new project since it is experimental and has no support for spigot or paper plugins

**Comment by encode42** [comment, marked as answer]
Maintaining a fork of Purpur based on Folia isn't worth the tradeoffs. Folia's target audience is those who can afford to allocate double-triple digits of threads to their servers. The type of networks that can afford to hire their own server and plugin developers (which would be needed anyway due to Folia's plugin breakage). If a network wants Purpur's patches, then they are free to implement them themselves.

Since Folia isn't for the average player, which *is* Purpur's target audience, it doesn't make much sense to work on something like this. At least not unless Folia becomes more common-place. (which likely isn't to happen.) Our audience simply cannot use Folia, because the average player typically doesn't have a big budget for the hardware they'd need.

Not to mention, us developers are busy, and as such, Purpur development has been slow. Creating yet another thing to maintain will only make this situation worse.

**Reply by subtogash** [reply]
How to add purpur pathces to folia?

**Comment by AndreyKrasnov06** [comment]
Good developers like you are sorely missed in the folia community. I'm using folia for my server which has only 10 threads allocated to it on my Ryzen 5 2600x. Folia does not have such a high barrier to entry. This is the future of all Minecraft servers. Unfortunately, very little folia forks are regularly updated and we, the creators of the servers, believe that developers like you will someday fix this.

**Reply by encode42** [reply]
Even if the Purpur developers wanted to port Purpur patches to Folia, it would essentially become an entirely separate project from Purpur. The amount of changes that would be required would effectively split it from mainline Purpur development, requiring double the resources to maintain and develop both projects. I doubt the Purpur team is willing to do this, as developers have left the project (such as me), meaning there's little overhead for additional projects. They would likely need to bring in new developers to exclusively develop for this new project and maintain parity with Purpur. Minecraft itself is in a similar situation with its two editions, which, despite their seemingly infinite money and resources, struggle to keep up with each other.

It's true that anyone can compile and run Folia, but that doesn't mean it's meant for everyone. Most consumer systems can't fully utilize the potential of Folia. It's still my belief that the (objectively) experimental aspect of Folia should only be involved with network developers that can afford to fund development to work around the quirks. Not to mention that the plugin ecosystem has still been split due to the fundamental breakages to the Bukkit API that come with multithreading the server.

I agree that Folia is definitely a step forward in the everlasting quest to turn Minecraft into an MMO, but it's such new and experimental software that the large amount of resources required to develop it can't be provided by a project as small as Purpur.

**Reply by Outfit8TSB** [reply]
well, the #1525 supposed to add a api of the tool which makes 2 editions meet in one server, Named Geyser, supporting The Plugin Version with extra tool as a spigot plugin named floodgate, making bedrock edition can play without java account.  purpur should've merged #1525 to access floodgate functions, but they closed it! i created pull request #1525 just to let purpur gain opportunity to let Bedrock to play! Not trolling purpur developers! why would i want to troll Purpur Developers?

**Reply by SytexMC** [reply]
@Outfit8TSB that has absolutely nothing to do with the topic?

**Comment by Outfit8TSB** [comment]
@joaoespanhol oh well, gotta work on 1.21 Radioactive eventhough the PlazmaMC didn't update, i will tell you after the Radioactive Updates.
you can use it as a alternative of purpurmc after i patched folia on radioactive. i will recomment on here when it's done.

**Comment by BillyGalbreath** [comment]
Since I like analogies, I'll put one here that I hope explains what Encode said in the accepted answer.

If we swapped Purpur with Cars, the Purpur team would be a bunch of amateur mechanics standing around in their home garage sipping beers working on a car for fun. What you guys are asking for here is a professional pit crew for a race car. If you guys can afford to have a race car, you don't want volunteer amateurs working on it. Hire an actual professional pit crew. It's all part of the cost of owning and operating a race car ;)
